File: Tipos de Dados/Matrizes Associativas/associativas.cpp
//C++
//Uso de biblioteca
#include <iostream>
#include <map>

using namespace std;

int main() {
    map<std::string, int> matriz;

    // Definição e inicialização
    matriz["um"] = 1;
    matriz["dois"] = 2;
    matriz["tres"] = 3;

    // Acessar elementos
    cout << "Valor para 'dois': " << matriz["dois"] << endl;

    // Remover um elemento
    matriz.erase("tres");

    // Iterar pela matriz
    for (const auto &par : matriz) {
        cout << par.first << ": " << par.second << endl;
    }

    return 0;
}

File: Tipos de Dados/Matrizes Associativas/Associativas.cs
//C Sharp (C#)
//Uso de Biblioteca

using System;
using System.Collections.Generic;

class Associativas {
    static void Main() {
        Dictionary<string, int> matriz = new Dictionary<string, int>();

        // Definição e inicialização
        matriz["um"] = 1;
        matriz["dois"] = 2;
        matriz["tres"] = 3;

        // Acessar elementos
        Console.WriteLine("Valor para 'dois': " + matriz["dois"]);

        // Remover um elemento
        matriz.Remove("tres");

        // Iterar pela matriz
        foreach (var par in matriz) {
            Console.WriteLine(par.Key + ": " + par.Value);
        }
    }
}
